src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +200,416 @@ public void recordTagValue(String time)
+            final int startPostionOfFloatValue = 4; // This is the first half of the second line
+            final int endPositionOfFloatValue = 8; // This is also in the first half of the second line
+            final int numBitsInBinary = 8;
+
+            /* This variable is used to find the last word of the file
+             * The line 'ends' in the middle of the line because the binary data is offset by half the line length
+             */
+            final int endPositionOfWordsWeNeed = lineWordLength / 2;
+
+           // Keep track of if we have read the entire binary file with this
+           readEntireFile = false;
+
+           while((singleCharInt = reader.read()) != -1) {
+
+               // Increase the word count since we just read in a new word
+               numWordRepresentedInDecimal++;
+
+               /* WhichLine is which line of data are we on. A line means one line of historical logging data
+                * MAX_DATA_POINTS_PER_TAG is for how many line should we read before giving up and going to the next tag
+                *
+                * This is one of the conditions to stop parsing the binary file and just move on to the next tag.
+                */
+               if(whichLine > MAX_DATA_POINTS_PER_TAG) {
+                   // Lines parsed has reached the max, stop parsing.
+                   Logger.LOG_DEBUG("Tag \"" + getTagName() + " no longer parsing file, max lin num hit. que size: " + dataPoints.size());
+                   reader.close();
+                   break;
+               }
+
+               /* addToQueue gets changed by data poster when removing points so we don't add and remove at the same time
+                *
+                * This is another condition to stop parsing and move on the the next tag.
+                */
+               if(!addToQueue) {
+                   Logger.LOG_DEBUG("Tag \"" + getTagName() + " parsing file stopped from dataposter while removing points");
+                   reader.close();
+                   readEntireFile = true;
+                   break;
+               }
+
+               // calculate start and end times of all binary data we want in units of word positions
+               int startPosTime = startPositionOfTimeStamp + whichLine * lineWordLength;
+               int endPosTime = endPositionOfTimeStamp + whichLine * lineWordLength;
+
+               int startPosValue = startPostionOfFloatValue + whichLine * lineWordLength;
+               int endPosValue = endPositionOfFloatValue + whichLine * lineWordLength;
+
+               /* Grab the 4 words in decimal and convert to binary.
+                * Once in binary, concatenate them all together to get the binary value
+                *
+                * Convert from binary back to decimal, and we have the epoch time in seconds.
+                */
+               if(numWordRepresentedInDecimal > startPosTime && numWordRepresentedInDecimal <= endPosTime  ) {

```

Comment:
definatly would like to see () around ((numWordRepresentedInDecimal > startPosTime) && (numWordRepresentedInDecimal <= endPosTime))

Also extra spaces after endPosTime